runtime/src/main/cpp/Natives.cpp

```diff
@@ -49,7 +49,7 @@ OBJ_GETTER0(Kotlin_native_internal_undefined) {
 }
 
 void* Kotlin_interop_malloc(KLong size, KInt align) {
-  if (size > SIZE_MAX) {
+  if (size < 0 || static_cast<size_t>(size) > std::numeric_limits<size_t>::max()) {
```

Comment:
Is this correct?
`KLong` might be longer than `size_t`, so `static_cast<size_t>(size)` would drop significant bits in this case, wouldn't it?

Reply:
That was me being silly. Added a cast to an appropriate unsigned type.